Meta/CMake/time_zone_data.cmake

```diff
@@ -38,30 +38,15 @@ set(TZDB_SOUTH_AMERICA_PATH "${TZDB_PATH}/${TZDB_SOUTH_AMERICA_SOURCE}")
 set(TZDB_ZONE_1970_SOURCE zone1970.tab)
 set(TZDB_ZONE_1970_PATH "${TZDB_PATH}/${TZDB_ZONE_1970_SOURCE}")
 
-function(extract_tzdb_file source path)
-    if(EXISTS "${TZDB_ZIP_PATH}" AND NOT EXISTS "${path}")
-        message(STATUS "Extracting TZDB ${source} from ${TZDB_ZIP_PATH}...")
-        execute_process(COMMAND "${TAR_TOOL}" -C "${TZDB_PATH}" -xzf "${TZDB_ZIP_PATH}" "${source}" RESULT_VARIABLE tar_result)
```

Comment:
keep this function around, and and wrap the execute_process and version check in a version check for 3.18

we can use the pattern `file(ARCHIVE_EXTRACT INPUT "${zip_path}" DESTINATION "${dest_dir}" PATTERNS "${source_path}")

to do the same thing that the manual extraction is doing. same for the unicode file.

Reply:
We have two options:

1. For unix targets, cmake 3.16 can be used (I will revert code), and for Windows - 3.18 will be needed.
2. Bump cmake requirement 3.18, and I will also restore the functionality to extract single files.

I propose a vote: thumbs up, means 1, down means 2. I will follow what ever you decide.